src/client/pages/Section/DataTable/Table/RowData/Cell/History/History.tsx

```diff
@@ -15,35 +19,55 @@ import { PropsCell } from '../props'
 
 type Returned = Array<Change>
 
-const useChanges = (props: { nodeValueA: NodeValue; nodeValueB: NodeValue }): Returned => {
-  const { nodeValueA, nodeValueB } = props
+const useIsNumeric = (col: Col) => {
+  return [ColType.integer, ColType.decimal, ColType.calculated].includes(col.props.colType)
+}
 
+const useChanges = (props: { nodeValueA: NodeValue; nodeValueB: NodeValue; row: Row; col: Col }): Returned => {
+  const { nodeValueA, nodeValueB, row, col } = props
+  const isNumeric = useIsNumeric(col)
   return useMemo<Returned>(() => {
-    const textA = nodeValueA?.raw ?? ''
-    const textB = nodeValueB?.raw ?? ''
+    const formatValue = (nodeValue: NodeValue, format?: { integer?: boolean }) => {
+      if (Objects.isEmpty(nodeValue?.raw)) return ''
+      if (isNumeric) {
+        const bigNumber = Numbers.toBigNumber(nodeValue.raw)
+        return Numbers.format(bigNumber, format?.integer ? 0 : 2).toString()
+      }
+      return nodeValue?.raw ?? ''
+    }
 
-    const changes = Diff.diffWords(textA, textB, {
-      ignoreCase: false,
-    })
+    const textA = formatValue(nodeValueA, row.props?.format)
+    const textB = formatValue(nodeValueB, row.props?.format)
 
-    return changes
-  }, [nodeValueA.raw, nodeValueB.raw])
+    return Diff.diffLines(textA, textB, { ignoreCase: false })
+  }, [nodeValueA, nodeValueB, row.props?.format, isNumeric])
 }
 
 const History: React.FC<PropsCell> = (props) => {
   const { nodeValue, table, col, row } = props
-  const { assessmentName, cycleName, countryIso } = useCountryRouteParams<CountryIso>()
+  const { assessmentName, /* cycleName: cycleNameParam, */ countryIso } = useCountryRouteParams<CountryIso>()
+  const assessment = useAssessment()
+  const cycle = useCycle()
   const tableName = table.props.name
   const { colName } = col.props
   const { variableName } = row.props
+  const isNumeric = useIsNumeric(col)
   const data = useLastApprovedHistoryTableData()
+
+  // TODO: Depending on HistoryLastApprovedInfo, cycleName is either current cycleName or prevCycleName
```

Comment:
Upcoming PRs when we use HistoryLastApprovedInfo for prevCycle vs lastApproved ts

Reply:
should we return previous cycle data under the current cycle key?